src/accounting/Consumer.cs

```diff
@@ -38,6 +58,7 @@ public void StartListening()
             {
                 try
                 {
+                    using var activity = MyActivitySource.StartActivity("order-consumed");
```

Comment:
somehow the db activity is not automatically closing. so i added this to group the child spans. otherwise, it would result in recursively nested spans like this 
![Uploading image.png…]()

Reply:
also have raised in the dotnet repo https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/2792

Reply:
thanks for pointing out! did not notice npsql also have native instrumentation. have updated accordingly and it is working as expected now 
![image](https://github.com/user-attachments/assets/d1e44e0b-9387-4898-85d8-966df1c563e3)